validator_client/src/http_api/mod.rs

```diff
@@ -1254,14 +1251,43 @@ where
                     .body(vec![])
                     .expect("can produce simple response from static values"),
             };
-
-            let body: &Vec<u8> = response.body();
-            let signature = signer(body);
-            let header_value =
-                HeaderValue::from_str(&signature).expect("hash can be encoded as header");
-
-            response.headers_mut().append("Signature", header_value);
-
             response
-        })
+        }
```

Comment:
This is bit isn't really doing what the function is advertised to do; I think it probably falls under the responsibility of the `blocking_signed_json_task` function instead as it describes here:

https://github.com/sigp/lighthouse/blob/f77a5a0a4e7bc565bb7079998eb81c2bdd85ccf4/validator_client/src/http_api/mod.rs#L1277-L1280

Is it possible to move this block there to keep the `convert_rejection` only do what it's supposed to do?

> Convert a warp `Rejection` into a `Response`.

If it's too much of an issue, we could consider renaming this function to something like `convert_into_json_response`?